### PR TITLE
[Issue #1823]: Fix prod FE error

### LINKED ISSUE #1823: [Task]: Fix prod FE error
### Summary

When visiting the search page with FF set:  https://simpler.grants.gov/search?_ff=showSearchV0:true

We get an error:
image

### Acceptance criteria

- FE degrades gracefully when API endpoints are not setup

### PR DESCRIPTION
**Pull request by rylew1**
## Summary
Fixes #1823 

## Changes proposed
- Add error handling when we have invalid JSON returned (when API is up but not endpoints are enabled)

image

Error we can simulate locally (same thing on prod):
```
Throwing error:  Not Found 404 {
  query: '',
  status: Set(0) {},
  fundingInstrument: Set(0) {},
  eligibility: Set(0) {},
  agency: Set(0) {},
  category: Set(0) {},
  sortby: null,
  page: 1,
  actionType: 'initialLoad'
}
Error fetching opportunities: T [Error]: {"type":"NotFoundError","searchInputs":{"query":"","status":[],"fundingInstrument":[],"eligibility":[],"agency":[],"category":[],"sortby":null,"page":1,"actionType":"initialLoad"},"message":"Unknown Error","status":404}
    at L (/Users/ryan/Desktop/simpler-grants-gov/frontend/.next/server/app/search/page.js:18:93101)
    at /Users/ryan/Desktop/simpler-grants-gov/frontend/.next/server/app/search/page.js:18:92757
    at H.sendRequest (/Users/ryan/Desktop/simpler-grants-gov/frontend/.next/server/app/search/page.js:18:92783)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async H.request (/Users/ryan/Desktop/simpler-grants-gov/frontend/.next/server/app/search/page.js:18:92406)
    at async H.searchOpportunities (/Users/ryan/Desktop/simpler-grants-gov/frontend/.next/server/app/search/page.js:18:93571)
    at async V.fetchOpportunities (/Users/ryan/Desktop/simpler-grants-gov/frontend/.next/server/app/search/page.js:18:94560)
    at async z (/Users/ryan/Desktop/simpler-grants-gov/frontend/.next/server/app/search/page.js:18:152457)
T [Error]: {"type":"NotFoundError","searchInputs":{"query":"","status":[],"fundingInstrument":[],"eligibility":[],"agency":[],"category":[],"sortby":null,"page":1,"actionType":"initialLoad"},"message":"Unknown Error","status":404}
    at L (/Users/ryan/Desktop/simpler-grants-gov/frontend/.next/server/app/search/page.js:18:93101)
    at /Users/ryan/Desktop/simpler-grants-gov/frontend/.next/server/app/search/page.js:18:92757
    at H.sendRequest (/Users/ryan/Desktop/simpler-grants-gov/frontend/.next/server/app/search/page.js:18:92783)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async H.request (/Users/ryan/Desktop/simpler-grants-gov/frontend/.next/server/app/search/page.js:18:92406)
    at async H.searchOpportunities (/Users/ryan/Desktop/simpler-grants-gov/frontend/.next/server/app/search/page.js:18:93571)
    at async V.fetchOpportunities (/Users/ryan/Desktop/simpler-grants-gov/frontend/.next/server/app/search/page.js:18:94560)
    at async z (/Users/ryan/Desktop/simpler-grants-gov/frontend/.next/server/app/search/page.js:18:152457)
[Error: An error occurred in the Server Components render. The specific message is omitted in production builds to avoid leaking sensitive details. A digest property is included on this error instance which may provide additional details about the nature of the error.] {
  digest: '1277226382'
}

```

### CODE REVIEW

File: frontend/src/app/api/BaseApi.ts [hunk 1]
```diff
@@ -222,6 +222,8 @@ const throwError = (
   searchInputs: SearchFetcherProps,
   firstError?: APIResponseError,
 ) => {
+  console.log("Throwing error: ", message, status_code, searchInputs);
+
```
Comment (andycochran) [inline frontend/src/app/api/BaseApi.ts:226]: Was this left in by mistake?

File: frontend/src/app/search/error.tsx [hunk 2]
```diff
@@ -27,7 +27,30 @@ export interface ParsedError {
...
+        sortby: null,
+        page: 1,
+        actionType: "initialLoad",
+      },
+      message: "Invalid JSON returned",
+      status: -1,
+    };
+  }
+
+  parsedErrorData = JSON.parse(error.message) as ParsedError;
```
Comment (andycochran) [inline frontend/src/app/search/error.tsx:53]: This won't set the var to falsey?

**Review by andycochran — APPROVED** [review]

**Comment by rylew1** [comment]
I had thought we might be able to just pass the error status code and type to the client error boundary component (`error.tsx`) - but it appears in a built version of the site it won't even let you pass the error message properly (or at least I need to do some further investigation as to why)

**Review by coilysiren — APPROVED** [review]
Approving again since it looks like Andy originally approved earlier

**Pull request merged**